### DISCUSSION TITLE
`xr.Dataset` with varying sizes of `xr.DataArrays`

### DISCUSSION THREAD

**Discussion by muaali** [pydata/xarray#5595] [Q&A]
Hi,

I am wondering if it is a good design choice to have an `xr.Dataset` that has `xr.DataArrays` of varying sizes ? Could it cause any issues with Parallelization with `dask`. Is this a bad design ?

My use case is to try to integrate values (x) with labels (y) in a single dataset (It doesn't have to be a single dataset though). So a user can easily fetch data samples with associated labels in an `xr.Dataset`
```
# da_values of dim (r,c,d)
# da_labels of dim (0,0,d), I am thinking to ingest labels in attributes. 
ds = xr.Dataset({"values": da_values, "labels": da_labels})
```

I am not sure if this is the most elegant solution and it does not look to me one. Could you please advise, comment. Thanks.

**Comment by max-sixty** [comment]
I frequently do this! One of my favorite features of xarray is I can use a dataset to encapsulate all the data I'm working with — rather than n dataframes with uncertain alignment. 

I'm not sure if there are dask issues — I wouldn't imagine they'd be severe, but I'll defer to others if they have thoughts.
